### PR TITLE
Fix another sign comparison case ...

### PR DESCRIPTION
**Pull request by Martin2112**
... and control reaching end of non void function warning.

### CODE REVIEW

File: cpp/server/ct-mirror.cc [hunk 1]
```diff
@@ -268,13 +268,19 @@ void STHUpdater(
         {
           LoggedCertificate entry;
           CHECK_LE(next_sth.tree_size(), local_size);
-          while (new_tree->LeafCount() < next_sth.tree_size()) {
+          CHECK_LE(next_sth.tree_size(), INT64_MAX);
```
Comment (AlCutter) [inline cpp/server/ct-mirror.cc]: I guess this should be `CHECK_GE(..., 0)`?

File: cpp/server/ct-mirror.cc [hunk 2]
```diff
@@ -268,13 +268,19 @@ void STHUpdater(
...
+          const uint64_t next_sth_tree_size(
+              static_cast<uint64_t>(next_sth.tree_size()));
+          while (new_tree->LeafCount() < next_sth_tree_size) {
             CHECK(entries->GetNextEntry(&entry));
             CHECK(entry.has_sequence_number());
-            CHECK_EQ(new_tree->LeafCount(), entry.sequence_number());
+            CHECK_LE(entry.sequence_number(), INT64_MAX);
```
Comment (AlCutter) [inline cpp/server/ct-mirror.cc]: >=0 ?

File: cpp/server/ct-mirror.cc (hunk 1 above)
Comment (Martin2112) [inline cpp/server/ct-mirror.cc]: Done.

File: cpp/server/ct-mirror.cc (hunk 2 above)
Comment (Martin2112) [inline cpp/server/ct-mirror.cc]: Done.

**Pull request merged**